src/rule_tester.rs

```diff
@@ -0,0 +1,150 @@
+use std::{fs, process::Command};
+
+use assert_cmd::prelude::*;
+use predicates::prelude::*;
+use tempdir::TempDir;
+
+use crate::rule::Rule;
+
+pub struct RuleTester {
+    rule: Rule,
+    rule_tests: RuleTests,
+}
+
+impl RuleTester {
+    fn new(rule: Rule, rule_tests: RuleTests) -> Self {
+        Self { rule, rule_tests }
+    }
+
+    pub fn run(rule: Rule, rule_tests: RuleTests) {
+        Self::new(rule, rule_tests).run_tests()
+    }
+
+    fn run_tests(&self) {
+        for valid_test in &self.rule_tests.valid_tests {
+            self.run_valid_test(valid_test);
+        }
+
+        for invalid_test in &self.rule_tests.invalid_tests {
+            self.run_invalid_test(invalid_test);
+        }
+    }
+
+    fn run_valid_test(&self, valid_test: &RuleTestValid) {
+        let tmp_dir = TempDir::new("valid_test").unwrap();
+        let test_filename = "tmp.rs";
+        fs::write(tmp_dir.path().join(test_filename), &valid_test.code).unwrap();
+        Command::cargo_bin("tree-sitter-lint")
```

Comment:
Similar to what I ran into elsewhere (`tree-sitter-grep`?) I was seeing this (from `assert_cmd`) using stale binaries so per its docs (a) first tried using `escargot` (and again ran into it hanging) and then (b) moved the tests from a unit-style-test into an integration-style test

I think these tests are "cute" as unit tests though so would be nice to figure out how to wrangle this in that environment or whatever